driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -638,6 +659,14 @@ public void onSet(Connection connection, Message.Response response, long latency
                         case RESULT:
                             if (((Responses.Result)response).kind == Responses.Result.Kind.PREPARED) {
                                 logger.debug("Scheduling retry now that query is prepared");
+                                Responses.Result.Prepared prepared = (Responses.Result.Prepared) response;
```

Comment:
It isn't necessary to retry the statement, because we could get the result from the previous execution and just provide a new metadata in the fresh PreparedStatement.